wordpress-mobile/WordPress-Android
# Add logging for all known migration errors

Fixes https://github.com/wordpress-mobile/WordPress-Android/issues/17746

Related PR: https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/119

To test:

1. Install the WordPress app (but do not log in)
2. Install the Jetpack app
3. Run the Jetpack app and observe logcat
4. Expect that the failure to migrate will be logged as an error with the tag "JETPACK_MIGRATION"

Bonus:

Manually cause a failure by emitting an error from a different migration step (e.g. sites)

Expect the error to be logged as well.

## Regression Notes
1. Potential unintended areas of impact
Migration

5. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and automated tests (including migration unit tests)

6. What automated tests I added (or what prevented me from doing so)
Adding tests for `AppLog` and `AppLogWrapper` proved to be time-consuming and problematic. The `AppLog` used in unit tests is different from the one used in production (to avoid calls to Android Framework methods), and the wrapper is not resolving correctly, resulting in binary incompatibility. It might be worth improving the architecture of the logging wrapper(s) to make adding tests more ergonomic. Currently, it seems that only mocking is supported, but actually verifying anything on those mocks results in classpath related build errors.

**Update**: I found a solution to the `verify` issue by providing a simple test double for `AppLogWrapper`. Usually, a wrapper would allow mocking (and verifying) directly, but this is needed to satisfy the type bindings in the test environment because we are already doing the same for `AppLog` (a shared dependency).

PR submission checklist:
